File: Questions/283. Move Zeros/main.py
def move_zeros(nums: list[int]) -> None:
    n = len(nums)
    read = write = 0
    while read < n:
        if nums[read] != 0:
            nums[write], nums[read] = nums[read], nums[write]
            write += 1
        read += 1

File: Questions/283. Move Zeros/main.cpp
#include<iostream>
using namespace std;

class Solution{
    public:
        void moveZeroes(vector<int>& nums) {
            int n = nums.size();
            int read = 0, write = 0;
            while (read < n) {
                if (nums[read] != 0) {
                    swap(nums[read], nums[write++]);
                }
                read++;
            }
        }
};